common-kafka/src/main/java/com/cerner/common/kafka/consumer/ProcessingKafkaConsumer.java

```diff
@@ -194,6 +195,21 @@ public ProcessingKafkaConsumer(ProcessingConfig config) {
         this(config, new KafkaConsumer<>(config.getProperties()));
     }
 
+    /**
+     * Creates a new {@link ProcessingKafkaConsumer} with the given configuration and deserializers using the {@link KafkaConsumer}.
+     *
+     * @param config the configuration used by the consumer
+     * @param keyDeserializer The deserializer instance for key. When passing the deserializer instance
+     *          directly, the {@link Deserializer#configure(Map, boolean)} method will not be called.
+     * @param valueDeserializer The deserializer instance for value. When passing the deserializer instance
+     *          directly, the {@link Deserializer#configure(Map, boolean)} method will not be called.
+     * @throws NullPointerException if config is {@code null}
+     * @throws KafkaException if there is an issue creating the {@link KafkaConsumer}
+     */
```

Comment:
add @throws javadoc about what to expect if either deserializer is null, and unit tests

Reply:
If the deserializers are null it will fall back to the `key.deserializer` and `value.deserializer` config properties. I updated the javadoc to indicate this, but didn't add any tests. If tests are still desired I could probably write a custom deserializer that increments a counter when `configure` is called and use the counter value for assertions.

https://github.com/cerner/common-kafka/pull/8/commits/291376e1fb68c4fb2fd3ae0ed9cf68a338fda504#diff-4f279fbd21bcd74432758ffce809d692R204

Reply:
looks good